docs/index.md

```diff
@@ -31,64 +31,130 @@ You can easily integrate it with your existing tools, and feel lakeFS in action
+
+<img src="{{ site.baseurl }}/assets/img/promotion_workflow.png" alt="lakeFS promotion workflow" width="60%" height="60%" />
+
+1. First, create a new **branch** sourced from main to instantly generate a complete copy of your production data.
+2. Apply changes or make updates on the isolated branch to understand their impact prior to exposure.
+3. And finally, perform a **merge** from the feature branch back to main to atomically promote the updates into production.
+
+Following this pattern, lakeFS facilitates a streamlined data deployment workflow that consistently produces data assets you can have total confidence in.
+
+## What else does lakeFS do?
+
+lakeFS helps you maintain a tidy data lake in several other ways, including:
+
+#### Recovery from data errors
+Erroneous data that makes it into production is a newar inevitability given the complex and fast-moving nature of modern data pipelines. Similarly, critical data assets are liable to accidental deletion by poorly configured jobs or careless humans.
+
+Today, recovering from these events relies on periodic backups that 1) may be out of date and 2) require sifting through data at the object level. This process is inefficient and can take hours, days, or in some cases, weeks to complete.
+Recovering data in deletion events becomes an instant one-line operation with lakeFS using the ability to restore any historical commit.
+
+Reverting your data lake back to previous version using our command-line tool is explained [here](https://docs.lakefs.io/reference/commands.html#lakectl-branch-reset).
+
+#### Data reprocessing and backfills
+
+Occasionally, we might need to reprocess historical data. This can be due to several reasons:
+
+* Implementing new logic.
+* Late arriving data that wasn’t included in previous analysis, and need to be backfilled after the fact.
 
-[Learn more](./usecases/production.md){:id="user-content-learn-more-dep"}
+This is tricky first and foremost because it often involves huge volumes of historical data. In addition, auditing requirements may necessitate keeping the old version of the data handy. 
+
+lakeFS allows you to manage the reprocess on an isolated branch before merging to ensure the reprocessed data is exposed atomically. It also allows you to easily access via a commit’s ID the different versions of reprocessed data.
+
+#### Cross-collection consistency guarantees
+
+Data engineers typically have to implement custom logic in scripts in order to guarantee two or more data assets are updated synchronously. The lakeFS merge operation from one branch into another removes the need to implement this logic yourself.
+
+Instead, make updates to the desired data assets on a branch, and then utilize a lakeFS merge to atomically expose the data to downstream consumers.
+
+To learn more about atomic cross-collection updates, [this video](https://www.youtube.com/watch?v=9OsjUvk5UJU) describes the concept in more detail.
```

Comment:
This sends me out of the documentation. I would prefer if we can embed this video like we did in the quick start - if it's important enough, or replace it with a link inside the docs.

Reply:
understand it's not ideal at the moment. Going to leave it in for now until we produce a better doc page explaining cross-coll consistency.